buildchain/buildchain/docker_command.py

```diff
@@ -0,0 +1,214 @@
+        self.repository = repository
+        self.digest = digest
+
+    def __call__(self) -> Union[TaskError, bool]:
+        try:
+            DOCKER_CLIENT.images.pull(self.repository, tag=self.digest)
+        except (docker.errors.BuildError, docker.errors.APIError) as err:
+            return doit.exceptions.TaskError(err)
+        return True
+
+
+# The call method is not counter as a public method
+# pylint: disable=too-few-public-methods
+class SaveCommand:
+    """Docker save callable"""
+
+    def __init__(self, tag: str, save_path: Path):
+        self.tag = tag
+        self.save_path = str(save_path)
+
+    def __call__(self) -> Union[TaskError, bool]:
+        try:
+            to_save = DOCKER_CLIENT.images.get(self.tag)
+            image_stream = to_save.save()
+            with open(self.save_path, 'wb') as image_file:
+                for chunk in image_stream:
+                    image_file.write(chunk)
+        except (docker.errors.APIError, IOError) as err:
+            return TaskError(err)
+        return True
+
+
+class RunCommand:
+    """Docker run callable"""
+
+    RPMLINTRC_MOUNT : Mount = Mount(
+        target='/rpmbuild/rpmlintrc',
+        source=str(constants.ROOT/'packages'/'rpmlintrc'),
+        type='bind',
+        read_only=True
+    )
+    ENTRYPOINT_MOUNT : Mount = Mount(
+        target='/entrypoint.sh',
+        source=str(constants.ROOT/'packages'/'entrypoint.sh'),
+        type='bind',
+        read_only=True
+    )
+    _BASE_CONFIG = {
+        'hostname': 'build',
+        'mounts': [ENTRYPOINT_MOUNT],
+        'environment': {
+            'TARGET_UID': os.geteuid(),
+            'TARGET_GID': os.getegid()
+        },
+        'tmpfs': {'/tmp': ''},
+        'remove': True
+    }
+
+    def __init__(
+        self,
+        command:     List[str],
+        builder:     Any,
+        environment: Optional[Dict[str, Any]]=None,
+        mounts:      Optional[List[Mount]]=None,
+        tmpfs:       Optional[Dict[str, str]]=None,
+        run_config:  Optional[Dict[str, Any]]=None,
+        read_only:   bool=False
+    ):
+        self.command = command
+        self.builder = builder
+        self.environment = environment or {}
+        self.mounts = mounts or []
+        self.tmpfs = tmpfs or {}
+        self.run_config = run_config or self.base_config()
+        self.read_only = read_only
+
+    @classmethod
+    def bind_mount(
+        cls, source: Path, target: str, **kwargs: Any
+    ) -> Mount:
+        """Helper for read/write mounts."""
+        return Mount(
+            source=str(source),
+            target=target,
+            type='bind',
+            **kwargs
+        )
+
+    @classmethod
+    def bind_ro_mount(cls, source: Path, target: str) -> Mount:
+        """Helper for read-only mounts."""
+        return cls.bind_mount(source=source, target=target, read_only=True)
+
+    @classmethod
+    def base_config(cls) -> Dict[str, Any]:
```

Comment:
`base_config` isn't really informative here.

IIUC, the `base_config` run the build container with the entrypoint, so maybe rename it `builder_config`?

(probably need to rename `_BASE_CONFIG` as well)